CONTRIBUTE.md

```diff
@@ -1,6 +1,34 @@
 # Contribute
 
-How to hack:
+## Issue
+
+Before writing any code, please create an issue first that describes the problem
+you are trying to solve with alternatives that you have considered. A little bit
+of prior communication can save a lot of time on coding. Keep the problem as
+small as possible. If there are two problems, make two issues. We discuss the
+issue and if we reach an agreement on the approach, it's time to move on to a
+PR.
```

Comment:
This is great. Maybe even add something about that it sometimes is best to start with a discussion thread, to collaboratively figure out what the problem(s) is/are?

Reply:
I'll add it